### PR TITLE
boards: arm: add nrf52840_pca10090

### PR DESCRIPTION
**Pull request by lemrey**
Add support for `nrf52840_pca10090` and a default configuration for it in the `hci_uart` project.
CC: @carlescufi @ioannisg @SebastianBoe @anangl

note: board image is a dummy :)

### CODE REVIEW

File: samples/bluetooth/hci_uart/nrf52840_pca10090.conf [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# Connect the MCU interface pins to the nRF9160
+# We will instantiate UART1 on these pins
+CONFIG_BOARD_PCA10090_INTERFACE_0_MCU=y
+CONFIG_BOARD_PCA10090_INTERFACE_1_MCU=y
+
+# Use UART1 for HCI
+CONFIG_UART_1_NRF_UARTE=y
+CONFIG_UART_1_NRF_FLOW_CONTROL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_BT_CTLR_TO_HOST_UART_DEV_NAME="UART_1"
```
Comment (carlescufi) [inline samples/bluetooth/hci_uart/nrf52840_pca10090.conf]: Is this necessary, since this is part of the `.overlay`?

**Review by dbkinder — APPROVED** [review]
+1 for doc changes

File: boards/arm/nrf52840_pca10090/doc/nrf52840_pca10090.rst [hunk 2]
```diff
@@ -0,0 +1,164 @@
+.. _nrf52840_pca10090:
+
+nRF52840-PCA10090
+#################
+
+Overview
+********
+
+Zephyr applications use the nrf52840_pca10090 board configuration
+to run on the nRF52840 SoC on the PCA10090 Development Kit hardware.
```
Comment (ru-fu) [inline boards/arm/nrf52840_pca10090/doc/nrf52840_pca10090.rst]:
```suggestion
to run on the nRF52840 SoC on the nRF9160 DK (PCA10090) hardware.
```

**Review by SebastianBoe — APPROVED** [review]

File: boards/arm/nrf52840_pca10090/doc/nrf52840_pca10090.rst [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+* Segger RTT (RTT Console)
+* :abbr:`UART (Universal asynchronous receiver-transmitter)`
+* :abbr:`WDT (Watchdog Timer)`
+
+The nRF52840 SoC does not have any connection to the any of the LEDs,
+buttons, switches and Arduino pin headers on the PCA10090 board. It is, however,
+possible to route some of the pins of the nRF52840 SoC to the nRF9160 SiP.
+
+More information about the board can be found at the
+`Nordic Low power cellular IoT`_. The `Nordic Semiconductor document library`_
```
Comment (ru-fu) [inline boards/arm/nrf52840_pca10090/doc/nrf52840_pca10090.rst]:
```suggestion
`Nordic Low power cellular IoT`_ website. The `Nordic Semiconductor documentation library`_
```

File: boards/arm/nrf52840_pca10090/doc/nrf52840_pca10090.rst [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+It is possbile to configure the behavior of the board controller firmware by
+using Kconfig and editing its options under "Board options".
+
+
+References
+**********
+
+.. target-notes::
+.. _Nordic Low power cellular IoT: https://www.nordicsemi.com/Products/Low-power-cellular-IoT
+.. _Nordic Semiconductor document library: https://www.nordicsemi.com/DocLib
```
Comment (ru-fu) [inline boards/arm/nrf52840_pca10090/doc/nrf52840_pca10090.rst]:
```suggestion
.. _Nordic Semiconductor documentation library: https://www.nordicsemi.com/DocLib
```

File: boards/arm/nrf52840_pca10090/doc/nrf52840_pca10090.rst [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+possible to route some of the pins of the nRF52840 SoC to the nRF9160 SiP.
+
+More information about the board can be found at the
+`Nordic Low power cellular IoT`_. The `Nordic Semiconductor document library`_
+contains the processor's information and the datasheet.
+
+Hardware
+********
+
+PCA10090 DK has two external oscillators. The frequency of
```
Comment (ru-fu) [inline boards/arm/nrf52840_pca10090/doc/nrf52840_pca10090.rst]:
```suggestion
The nRF9160 DK has two external oscillators. The frequency of
```

File: samples/bluetooth/hci_uart/nrf52840_pca10090.overlay [hunk 6]
```diff
@@ -0,0 +1,15 @@
+/ {
+	chosen {
+		zephyr,bt-uart=&uart1;
+	};
+};
+
+&uart1 {
```
Comment (ioannisg) [inline samples/bluetooth/hci_uart/nrf52840_pca10090.overlay]: How do we solve the problem with uart1 being enabled in the nrf9160_pc10090 by default?

File: boards/arm/nrf52840_pca10090/doc/nrf52840_pca10090.rst [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+* :abbr:`NVIC (Nested Vectored Interrupt Controller)`
+* :abbr:`PWM (Pulse Width Modulation)`
+* RADIO (Bluetooth Low Energy and 802.15.4)
+* :abbr:`RTC (nRF RTC System Clock)`
+* Segger RTT (RTT Console)
+* :abbr:`UART (Universal asynchronous receiver-transmitter)`
+* :abbr:`WDT (Watchdog Timer)`
+
+The nRF52840 SoC does not have any connection to the any of the LEDs,
+buttons, switches and Arduino pin headers on the PCA10090 board. It is, however,
```
Comment (ru-fu) [inline boards/arm/nrf52840_pca10090/doc/nrf52840_pca10090.rst]:
```suggestion
buttons, switches, and Arduino pin headers on the PCA10090 board. It is, however,
```

File: boards/arm/nrf52840_pca10090/doc/nrf52840_pca10090.rst [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+with a Segger IC.
+
+Remember to set the PROG/DEBUG switch on the DK to nRF52.
+
+Board controller firmware
+*************************
+
+The board controller firmware is a small snippet of code that takes care of
+routing specific pins on nRF9160 SiP to different components on the board,
+such as LEDs, switches and specific nRF52840 SoC pins.
```
Comment (ru-fu) [inline boards/arm/nrf52840_pca10090/doc/nrf52840_pca10090.rst]:
```suggestion
such as LEDs, switches, and specific nRF52840 SoC pins.
```

File: samples/bluetooth/hci_uart/nrf52840_pca10090.conf (hunk 1 above)
Comment (lemrey) [reply samples/bluetooth/hci_uart/nrf52840_pca10090.conf]: I double checked: `CONFIG_BT_CTLR_TO_HOST_UART_DEV_NAME` will default to "UART_0" regardless of which UART is specified in the `dts`.

**Comment by lemrey** [comment]
@ru-fu Thank you, I have addressed your comments.

File: samples/bluetooth/hci_uart/nrf52840_pca10090.overlay (hunk 6 above)
Comment (ioannisg) [reply samples/bluetooth/hci_uart/nrf52840_pca10090.overlay]: Right, my mistake.

**Review by galak — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Pull request merged**